src/navigatorBreadcrumbs.jsx

```diff
@@ -17,12 +17,37 @@
  * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
  */
 import cockpit from "cockpit";
-import React from "react";
+import React, { useEffect, useState } from "react";
 
 import { Button, Flex, FlexItem, PageBreadcrumb } from "@patternfly/react-core";
 import { ArrowLeftIcon, ArrowRightIcon } from "@patternfly/react-icons";
 
 export const NavigatorBreadcrumbs = ({ currentDir, path, history, setHistory, historyIndex, setHistoryIndex }) => {
+    const [hostnameData, setHostnameData] = useState(undefined);
+
+    useEffect(() => {
+        const client = cockpit.dbus('org.freedesktop.hostname1');
+        const hostnameProxy = client.proxy('org.freedesktop.hostname1', '/org/freedesktop/hostname1');
+        hostnameProxy.addEventListener("changed", data => {
+            setHostnameData(data.detail);
+        });
+    }, []);
+
+    const hostnameText = () => {
+        if (!hostnameData)
+            return undefined;
+
+        const pretty_hostname = hostnameData.PrettyHostname;
+        const static_hostname = hostnameData.StaticHostname;
+        let str = hostnameData.Hostname;
+        if (pretty_hostname && static_hostname && static_hostname !== pretty_hostname)
+            str = pretty_hostname + " (" + static_hostname + ")";
+        else if (static_hostname)
+            str = static_hostname;
+
+        return str || '';
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-245-20240205-200740-ed73c12c-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)